src/zenml/zen_stores/schemas/artifact_schemas.py

```diff
@@ -104,7 +109,11 @@ class ArtifactSchema(NamedSchema, table=True):
     workspace: "WorkspaceSchema" = Relationship(back_populates="artifacts")
     run_metadata: List["RunMetadataSchema"] = Relationship(
         back_populates="artifact",
-        sa_relationship_kwargs={"cascade": "delete"},
+        sa_relationship_kwargs=dict(
+            primaryjoin=f"and_(RunMetadataSchema.resource_type=='{MetadataResourceTypes.ARTIFACT.value}', foreign(RunMetadataSchema.resource_id)==ArtifactSchema.id)",
```

Comment:
Same here, the `ArtifactSchema` will be split into two schemas - [PR here](https://github.com/zenml-io/zenml/pull/2081). This `Relationship` will be affected.